fiona/ogrext1.pyx

```diff
@@ -203,13 +203,18 @@ cdef class FeatureBuilder:
             elif fieldtype in (FionaDateType, FionaTimeType, FionaDateTimeType):
                 retval = ogrext1.OGR_F_GetFieldAsDateTime(
                     feature, i, &y, &m, &d, &hh, &mm, &ss, &tz)
-                if fieldtype is FionaDateType:
-                    props[key] = datetime.date(y, m, d).isoformat()
-                elif fieldtype is FionaTimeType:
-                    props[key] = datetime.time(hh, mm, ss).isoformat()
-                else:
-                    props[key] = datetime.datetime(
-                        y, m, d, hh, mm, ss).isoformat()
+                if not y or not m or not d:
+                    log.warning("Invalid date")
+                try:
+                    if fieldtype is FionaDateType:
+                         props[key] = datetime.date(y, m, d).isoformat()
+                    elif fieldtype is FionaTimeType:
+                         props[key] = datetime.time(hh, mm, ss).isoformat()
+                    else:
+                         props[key] = datetime.datetime(
+                             y, m, d, hh, mm, ss).isoformat()
+                except ValueError:
+                    props[key] = None
```

Comment:
... and log the actual exception that occurred, which will be something like `ValueError: year is out of range`, using https://docs.python.org/2/library/logging.html#logging.Logger.exception before setting the property value to `None`.